hadoop-ozone/dist/src/main/compose/testlib.sh

```diff
@@ -574,3 +574,35 @@ wait_for_root_certificate(){
   echo "Timed out waiting on $count root certificates. Current timestamp " $(date +"%T")
   return 1
 }
+
+execute_s3a_tests() {
+  local bucket="$1"
+
+  if [[ -z ${bucket} ]]; then
+    echo "Required argument: the S3 bucket to be tested" >&2
+    return 1
+  fi
+
+  if [[ -z ${HADOOP_AWS_DIR} ]] || [[ ! -e ${HADOOP_AWS_DIR} ]] || [[ ! -d ${HADOOP_AWS_DIR}/src/test/resources ]]; then
+    echo "Set HADOOP_AWS_DIR to the directory with hadoop-aws sources" >&2
+    return 1
+  fi
+
+  if [[ -z ${OZONE_S3G_ADDRESS} ]]; then
+    echo "Set OZONE_S3G_ADDRESS to the address of S3 Gateway" >&2
+    return 1
+  fi
+
+  pushd ${HADOOP_AWS_DIR}
+  mvn -B -V --no-transfer-progress \
+    -Dtest.fs.s3a.endpoint="${OZONE_S3G_ADDRESS}" \
+    -Dcustom.fs.s3a.name="s3a://${bucket}/" \
```

Comment:
Where are these properties read in the Hadoop code? I'm only seeing the test endpoint referenced [here](https://github.com/apache/hadoop/blob/7fb9c306e22eee334667d4b761e3fd330f45cb01/hadoop-tools/hadoop-aws/src/test/java/org/apache/hadoop/fs/s3a/ITestS3AConfiguration.java#L104). Also how do these values interact with auth-keys.xml?

Reply:
These were referenced in `auth-keys.xml`, [substituted](https://github.com/apache/hadoop/blob/7db9895000860605a66dd6403005b0c61a6ed744/hadoop-common-project/hadoop-common/src/main/java/org/apache/hadoop/conf/Configuration.java#L1103-L1130) by Hadoop `Configuration`.

Updated the patch to generate `auth-keys.xml` for each test bucket instead of relying on the substitution.  This allows running the tests when using the binary tarball (which does not include CI check scripts).